tensorflow/go/BUILD.bazel

```diff
@@ -0,0 +1,74 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+licenses(["notice"])  # Apache 2.0
+
+exports_files(["LICENSE"])
+
+sh_test(
+    name = "test",
+    size = "small",
+    srcs = ["test.sh"],
+    data = [
+        ":all_files",  # Go sources
+        "//tensorflow:libtensorflow.so",  # C library
+        "//tensorflow/c:headers",  # C library header
+        "//tensorflow/cc/saved_model:saved_model_half_plus_two",  # Testdata for LoadSavedModel
+    ],
+)
+
+filegroup(
+    name = "all_files",
+    srcs = glob(
+        ["**/*"],
+        exclude = [
+            "**/METADATA",
+            "**/OWNERS",
+        ],
+    ),
+    visibility = ["//tensorflow:__subpackages__"],
+)
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "doc.go",
+        "graph.go",
+        "lib.go",
+        "operation.go",
+        "saved_model.go",
+        "session.go",
+        "shape.go",
+        "status.go",
+        "tensor.go",
+        "version.go",
+        "//tensorflow/c:headers",  # C library header
+    ],
+    cgo = True,
+    clinkopts = ["-L/usr/local/lib -ltensorflow"],
```

Comment:
I know it's possible to `import cgo` and have source code comments that schlep in pkg-config. So could it do something like this?

```sh
TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )
g++ -std=c++11 -shared zero_out.cc -o zero_out.so -fPIC ${TF_CFLAGS[@]} ${TF_LFLAGS[@]} -O2
```

*Note: copied from https://www.tensorflow.org/extend/adding_an_op*

Out of curiosity, why have the go bindings in the TensorFlow repository, if it TensorFlow comes from the system? Hosting the code in this repo, it might make more sense to statically link TensorFlow. Or possibly have libframework.so etc. specified as `data = [...]` and then you can reference them when `bazel run`.